### PR TITLE
Changes to /docs

### PR DESCRIPTION
**Pull request by leekahung**
Thought about using CONTRIBUTING.md before, but I think this could also work.

To solve the problem of generated docs, I've updated .gitignore to ignore all other files in /docs except /docs/README.md. This will prevent everything, but /docs/README.md from being uploaded to GitHub in the future.

/docs/README.md is still important as an entry point for generating documentation locally for developers.

Other changes includes:
- The removal of all other files and directories from /docs
- Updated package.json to include the ability to locally clear all other files/directories from /docs via the "npm run docs:clear" command

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "build": "vite build",
     "preview": "vite preview",
     "docs": "jsdoc --readme ./docs/README.md -c jsdoc.config.json",
+    "docs:clear": "cd docs && mv README.md ../temp.md && rm -rf * && cd ../ && mv temp.md docs/README.md",
```
Comment (k-mack) [inline package.json]: This is good, but it won't work on Windows. Can you refactor this into a JS script instead? That should make it platform agnostic.

**Review by k-mack — CHANGES_REQUESTED** [review]
Looks great. Once we can clear the docs in a platform-agnostic way, this will be perfect.

File: package.json (hunk 1 above)
Comment (leekahung) [reply package.json]: Sure! I'll refactor this.

**Review by k-mack — APPROVED** [review]
Excellent!

**Comment by k-mack** [comment]
@leekahung I tweaked the target branch and that made your branch fall out of alignment with it. Can you rebase your branch onto `origin/issue-20/add-build-system`:

```
# Checkout this branch.
git checkout issue-20/add-build-system-suggestions

# Fetch changes pushed to remote `origin` server.
git fetch origin

# Rebase this branch (issue-20/add-build-system-suggestions) onto the server's `issue-20/add-build-system` branch.
git rebase origin/issue-20/add-build-system
```

When it attempts the rebase, git will pause and ask you to resolve the conflict in `package.json`. Just you need to add `docs/` to the `--readme` option in the `docs` script. After you resolve the conflict, do a `git add package.json` followed by a `git rebase --continue` to continue the rebase operation. It should complete successfully and the merge conflicts should be resolved. Then, force push your branch.

Sorry for the headache!

**Comment by leekahung** [comment]
No worries! I got it done. Looks ready for merging @k-mack.

**Pull request merged**